### PR TITLE
Patch in CCustomScript::ResolvePath for ModLoader support.

### PR DESCRIPTION
**Pull request by MiranDMC**

### CODE REVIEW

**Review by x87 — CHANGES_REQUESTED** [review]
please delete commented code

**Comment by MiranDMC** [comment]
> please delete commented code

This is that check for safe paths. This should be implemented in correct way instead of just disabling.

**Comment by x87** [comment]
> > please delete commented code
> 
> This is that check for safe paths. This should be implemented in correct way instead of just disabling.

are you going to update it in this PR?

File: cleo_sdk/CLEO_Utils.h [hunk 1]
```diff
@@ -117,6 +117,38 @@ namespace CLEO
         return std::move(info);
     }
 
+    // does file path points inside game directories? (game root or user files)
+    static bool IsFilepathSafe(CLEO::CRunningScript* thread, const char* path)
+    {
+        // TODO: path handling should use more advanced utils from std::filesystem
```
Comment (x87) [inline cleo_sdk/CLEO_Utils.h:123]: maybe call [canonical](https://en.cppreference.com/w/cpp/filesystem/canonical) first, then check if the path starts with root/userfiles ?

Comment (MiranDMC) [reply cleo_sdk/CLEO_Utils.h:123]: Yes, but this will need including std::filesystem. I'm not sure if this header should create that dependency.

Comment (x87) [reply cleo_sdk/CLEO_Utils.h:123]: can you move it to a separate header file then?

Comment (MiranDMC) [reply cleo_sdk/CLEO_Utils.h]: updated

Comment (x87) [reply cleo_sdk/CLEO_Utils.h]: OK, I'm not familiar with filesystem APIs in C++, so I'm going to trust you that this is working as intended.

**Comment by MiranDMC** [comment]
Still what with Skin Loader killing other scripts?
https://github.com/cleolibrary/CLEO5/pull/143/commits#diff-6ee87db9cf9c4b2cdb96aaaaafbdcfee97e54deb2e1fbe93d66f90576c0b0e6a

**Comment by x87** [comment]
This is what it also does in CLEO4, so users should already suffer from it. We can discuss potential mitigation, but it's not our primary concern right now.

**Pull request merged**